File: Java/src/test/java/codekata/DeviceDriverTest.java
package codekata;

import static org.junit.Assert.*;

import org.junit.Test;


public class DeviceDriverTest {

    @Test
    public void read_From_Hardware() {
        // TODO: replace hardware with a Test Double
        FlashMemoryDevice hardware = null;
        DeviceDriver driver = new DeviceDriver(hardware);
        byte data = driver.read(0xFF);
        assertEquals(0, data);
    }
}

File: Kotlin/src/test/kotlin/codekata/DeviceDriverTest.kt
package codekata

import org.junit.Assert.assertEquals
import org.junit.Test

internal class DeviceDriverTest {

    @Test
    fun read_From_Hardware() {
        // TODO: replace hardware with a Test Double
        val hardware: FlashMemoryDevice = null!!
        val driver = DeviceDriver(hardware)
        val data = driver.read(0xFF)
        assertEquals(0, data.toLong())
    }

}
